pathway-analysis/02-gene_set_enrichment_analysis.Rmd

```diff
@@ -111,77 +126,121 @@ dge_results_df <- readr::read_tsv(dge_results_file)
+
+```{r gene_dups, live = TRUE}
+duplicated_gene_symbols <- dge_results_df %>%
+  dplyr::filter(duplicated(gene_symbol)) %>%
+  dplyr::pull(gene_symbol)
+```
 
-When `mapIds()` is run with `multiValues = "first"` and it encounters multiple matches, only the first identifier is returned.
-This is the default behavior.
-It will also return a named vector of the IDs you queried for (`column`), where the names are the input identifiers (`keys`), in the same order as the keys which means you can use it directly with `dplyr::mutate()`.
+Now we'll look at the values for the the duplicated gene symbols.
 
-```{r convert_entrez, live = TRUE}
-dge_results_df <- dge_results_df %>%
-  # Create a new column 'entrez_id' that contains the Entrez IDs returned by
-  # mapIds()
-  dplyr::mutate(entrez_id = mapIds(org.Hs.eg.db, 
-                                   keys = dge_results_df$Gene, 
-                                   column = "ENTREZID", 
-                                   keytype = "ENSEMBL", 
-                                   multiVals = "first")
-  ) %>%
-  # We need these gene identifiers to perform our pathway analysis, so remove
-  # any genes where we don't have Entrez IDs
-  dplyr::filter(!is.na(entrez_id))
+```{r show_gene_dups}
+dge_results_df %>%
+  dplyr::filter(gene_symbol %in% duplicated_gene_symbols) %>%
+  dplyr::arrange(gene_symbol)
 ```
 
-### Preranked list
+We can see that the associated values vary for each row.
 
-The `GSEA()` function takes a preranked (sorted) named vector of statistics, where the names in the vector are gene identifiers.
+Let's keep the gene symbols associated with the higher absolute value of the log2 fold change.
+GSEA relies on genes' rankings on the basis of a gene-level statistic and the enrichment score that is calculated reflects the degree to which genes in a gene set are overrepresented in the top or bottom of the rankings ([Subramanian _et al._ 2005](https://doi.org/10.1073/pnas.0506580102); [Yu](http://yulab-smu.top/clusterProfiler-book/chapter2.html#gene-set-enrichment-analysis)).
+
+Retaining the instance of the gene symbols with the higher absolute value of a gene-level statistic means that we will retain the value that is likely to be more highly- or lowly-ranked or, put another way, the values less likely to be towards the middle of the ranked gene list.
+We should keep this decision in mind when interpreting our results.
+For example, if all the duplicate identifiers happened to be in a particular gene set, we may get an overly optimistic view of how perturbed that gene set is because we preferentially selected instances of the identifier that have a higher absolute value of the statistic used for ranking.
+
+In the next chunk, we are going to filter out the duplicated rows using the `dplyr::distinct()` function
+This will keep the first row with the duplicated value thus keeping the row with the highest absolute value of the log2 fold change.
```

Comment:
Slight clarification (because a period was missing).
```suggestion
In the next chunk, we are going to filter out the duplicated rows using the `dplyr::distinct()` function after sorting by absolute value of the log2 fold change.
This will keep the first row with the duplicated value thus keeping the row with the largest absolute value.
```